File: multi-lang/aoc2022/1/part1/zig/src/main.zig
const std = @import("std");

const input = @embedFile("input");

pub fn main() anyerror!void {
    var lines = std.mem.split(u8, input, "\n");

    var highest: u32 = 0;
    var current: u32 = 0;

    while (lines.next()) |line| {
        if (line.len == 0) {
            if (current >= highest) {
                    highest = current;
                }

            current = 0;
            continue;
        }

        var parsed = try std.fmt.parseInt(u32, line, 0);
        current += parsed;
    }

    std.log.info("Highest = {d}", .{highest});
}

File: multi-lang/aoc2022/1/part1/rust/src/main.rs
// fn main() {
//     let input = include_str!("input");

//     let mut current: u32 = 0;
//     let mut highest: u32 = 0;

//     for line in input.split("\n") {
//         if line.len() == 0 {
//             if current >= highest {
//                 highest = current;
//             }

//             current = 0;
//             continue;
//         }

//         let parsed: u32 = line.parse().unwrap();
//         current += parsed;
//     }

//     println!("Highest = {}", highest);
// }

// small rust
fn main() {
    let input = include_str!("input");

    let highest: u32 = input
        .split("\n\n")
        .map(|group| {
            group
                .split("\n")
                .map(|line| line.parse::<u32>().unwrap_or(0))
                .sum()
        })
        .max()
        .unwrap();

    println!("Highest = {}", highest);
}
